### DISCUSSION TITLE
Import in getServerSideProp ends up in client bundle

### DISCUSSION THREAD

**Discussion by mderazon** [vercel/next.js#53487] [Help] [answered]
### Summary

I am investigating why I see `crypto-browserify` polyfill in my client bundle. 
In my code, I have something like this

```tsx
// /pages/[myPage].tsx
import { crypto } from 'crypto';

export const getServerSideProps: GetServerSideProps<MyProps> = async (context) => {
  // use crypto package here
}

// not using crypto anywhere else
```

In this case, crypto import is done at the top of the page file, but is only used within `getServerSideProps` context.
Will it still end up in the client bundle ?

**Comment by icyJoseph** [comment]
Buffer, crypto, stream, those are usually culprits as to why crypto-browserify ends up in your bundle. 

I'd say, just import crypto inside GSSP, and don't sweat it. Here I had made a similar issue, https://github.com/icyJoseph/next-js-react-gbg/blob/47d4cdaa070ad21ca0160fb347eacccaf4e17d9c/demo-app/src/lib/pokemon.ts, a string exported from a module that imports crypto.

Could you also double check that this is not only because of that crypto import? Like, if you remove it, do you still see crypto-browserify in the final bundle?

**Reply by mderazon** [reply]
You're saying that `fetchPokemon` in GSSP resulted in crypto being added to the main bundle, even though crypto is being used in `catchPokemon` ? 
How did you resolve ?

**Reply by icyJoseph** [reply]
Nope, in my case it was the string for the endpoint, that was used in client code. That brought crypto-browserify into my bundle. 

To solve that, I moved the string into its own module.

**Comment by mderazon** [comment, marked as answer]
Next.js eliminates imports in GSSP, you can use this tool to see what will included in client bundle
https://next-code-elimination.vercel.app/

https://nextjs.org/docs/pages/building-your-application/data-fetching/get-server-side-props#when-does-getserversideprops-run
